src/main/java/Ran.java

```diff
@@ -0,0 +1,231 @@
+        switch (type) {
+        case TODO:
+            // Take string after "todo" word
+            description = input.substring(5);
+            list[listCount] = new Todo(description);
+            break;
+        case DEADLINE:
+            int byIndex = input.indexOf("/by");
+            if (byIndex == -1) {
+                throw new MissingArgumentException(CommandType.DEADLINE);
+            }
+            // Take string between "deadline" and "/by"
+            description = input.substring(9, byIndex - 1);
+            // Take string after "/by"
+            String by = input.substring(byIndex + 4);
+            list[listCount] = new Deadline(description, by);
+            break;
+        case EVENT:
+            int fromIndex = input.indexOf("/from");
+            int toIndex = input.indexOf("/to");
+            if (fromIndex == -1 || toIndex == -1) {
+                throw new MissingArgumentException(CommandType.EVENT);
+            }
+            // Take string between "event" and "/from"
+            description = input.substring(6, fromIndex - 1);
+            // Take string between "/from" and "/to"
+            String from = input.substring(fromIndex + 6, toIndex - 1);
+            // Take string after "/to"
+            String to = input.substring(toIndex + 4);
+            list[listCount] = new Event(description, from, to);
+            break;
+        case UNDEFINED:
+        default:
+            list[listCount] = new Task(input);
+        }
+        listCount++;
+        printAddedTask();
+    }
+
+    public static void showList() throws EmptyListException {
+        if (listCount == 0) {
+            throw new EmptyListException();
+        }
+        System.out.println(LINE);
+        for (int i = 0; i < listCount; i++) {
+            System.out.println("\t" + (i + 1) + "." + list[i]);
+        }
+        System.out.println(LINE);
+    }
+
+    public static void markTask(String taskNum) throws OutOfListBoundsException {
+        int taskNumber = Integer.parseInt(taskNum) - 1;
+        if (taskNumber >= listCount || taskNumber < 0) {
+            throw new OutOfListBoundsException();
+        }
+        list[taskNumber].setAsDone();
+        System.out.println(LINE);
+        System.out.println("\tNice! I've marked this task as done:");
+        System.out.println("\t  " + list[taskNumber]);
+        System.out.println(LINE);
+    }
+
+    public static void unmarkTask(String taskNum) throws OutOfListBoundsException {
+        int taskNumber = Integer.parseInt(taskNum) - 1;
+        if (taskNumber >= listCount || taskNumber < 0) {
+            throw new OutOfListBoundsException();
+        }
+        list[taskNumber].setAsUndone();
+        System.out.println(LINE);
+        System.out.println("\tOK, I've marked this task as not done yet:");
+        System.out.println("\t  " + list[taskNumber]);
+        System.out.println(LINE);
+    }
+    
+    // Read user input for command, throw exception for invalid commands
```

Comment:
Good job on commenting your code. However, I would recommended to use Javadoc format for method comments instead